### DISCUSSION TITLE
How to add a new item to DataGridCollectionView

### DISCUSSION THREAD

**Discussion by buminduyasith97** [AvaloniaUI/Avalonia#10528] [Q&A]
```
  [ObservableProperty]
  private DataGridCollectionView? azureDevopsWorkItemErrors;

public Result(){
   List<AzureDevopsWorkItemErrorsDto> xx = new List<AzureDevopsWorkItemErrorsDto>()
            {
                new AzureDevopsWorkItemErrorsDto {RowId = 1,Error = "error1"},
                new AzureDevopsWorkItemErrorsDto {RowId = 2 , Error = "error 2"},
            };

            AzureDevopsWorkItemErrors = new DataGridCollectionView(xx);
}

public AddnewItem(){

      var x = new AzureDevopsWorkItemErrorsDto { RowId = 3, Error = "error 3" };
      AzureDevopsWorkItemErrors.AddNew()
}

```

this "Addnew" method does not take any  arguments. can you please explain me how to add a new item to the grid

**Comment by timunie** [comment]
Looking at the source here I think the method returns the new object which you can manipulate afterwards: https://github.com/AvaloniaUI/Avalonia/blob/3c1177ced572261b1d14f393eaec03c300ad46e7/src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs#L1163

Otherwise you need to add the item to your source collection directly iirc.

As an alternative you can also use an `ObservableCollection` as the collection to bind or use `DynamicData` (if you don't need grouping).

**Reply by Snailya** [reply]
If I need grouping, how to?
I use dynamicdata to refresh the collection, but whenever there is a change, the group is removed, I want to keep that group.

**Comment by cedric56** [comment]
@timunie When i call this method i get an exception, and CanAddNew is always false. So what i am missing ?

Here is how i instanciate the collection

```
public DataGridViewModel()
{

    var persons = new ObservableCollectionSummarizable<PersonViewModel>(Repository.Instance.Persons.Select(p => new PersonViewModel(p)
    {
        Age = Random.Shared.Next(1, 99),
        Salary = Random.Shared.Next(1000, 2900),
        Sex = (SexTypes)Sexes.GetValue(Random.Shared.Next(Sexes.Length))

    }));

    CollectionView = new DataGridCollectionView(persons);
    CollectionView.SortDescriptions.Add(DataGridSortDescription.FromPath("Person.FirstName", ListSortDirection.Ascending, Comparer.Default));
    CollectionView.GroupDescriptions.Add(new DataGridPathGroupDescription("Age"));
    CollectionView.GroupDescriptions.Add(new DataGridPathGroupDescription("Sex"));

    //CollectionView.AddNew();
}
```

![image](https://github.com/user-attachments/assets/f189c000-a007-4d84-8449-6b32b473962b)

**Reply by timunie** [reply]
> [!NOTE]
> 
> 1. The underlaying collection must allow adding new items
> 2. The Items class must have a public parameterless constructor

**Reply by cedric56** [reply]
Thanks  for the response, it works. Where does this not come from ?

**Reply by timunie** [reply]
You can investigate the source since Avalonia is OpenSource. I think there is no explicit docs for this unfortunately. If you want to add some notes in the docs or XML API docs feel free to provide a PR
